### PR TITLE
arion: 0.1.2.0 -> 0.1.3.0

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Update arion to the new release that better supports 21.05.
Users may want to switch to podman https://docs.hercules-ci.com/arion/#_installation

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125563` at c8f09c28 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>arion</li>
    <li>haskellPackages.arion-compose</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/haskell-modules/generic-builder.nix:646:3:
```
    |
646 |   meta = { inherit homepage license platforms; }
    |   ^
```
</li>
  </ul>
</details>

**Comment by maralorn** [comment]
@cdepillabout @sternenseemann Do you have any issues with this approach?

I would be okay with this and will merge when no one objects.

**Comment by cdepillabout** [comment]
> Do you have any issues with this approach?

Huh, I'm surprised this works.  I didn't realize you could just completely override existing packages in `pkgs/development/haskell-modules/non-hackage-packages.nix`.

I'm also not too worried about technical debt on the release branches, but @roberth could you add a comment to `pkgs/development/haskell-modules/non-hackage-packages.nix` explaining what is going on with this override?  I imagine it would be confusing if someone stumbled over this.

**Comment by sternenseemann** [comment]
> Huh, I'm surprised this works. I didn't realize you could just completely override existing packages in `pkgs/development/haskell-modules/non-hackage-packages.nix`.

It works because the non hackage packages are added to the package set in the form of an overlay: https://github.com/NixOS/nixpkgs/blob/b382a5e61366a3e0b96945ced22fd24685c91cf4/pkgs/development/haskell-modules/default.nix#L31-L39

I think the approach makes a lot of sense for now, it is one that will always work regardless how much a package's cabal file changes between the version we have and the version we need.

**Review by sternenseemann — APPROVED** [review]

**Pull request merged**
